### PR TITLE
🧪 Add tests for smartPaste widget detection

### PR DESCRIPTION
**Pull request by OPS-PIvers**
This PR significantly improves the test coverage and reliability of the `smartPaste` utility.

### 🎯 What:
Addressed missing test cases for the `detectWidgetType` function in `utils/smartPaste.ts`.

### 📊 Coverage:
The following scenarios are now covered:
- **Input Sanitization:** Empty and whitespace-only strings correctly return `null`.
- **Share Links:** Both full URLs and bare domains are correctly identified for Boards, Quizzes, and Assignments.
- **Stickers:** Various image extensions (png, jpg, webp, svg) and URLs with query parameters or hashes are correctly detected.
- **Embeds:** Verified that Google Sheets and Forms are correctly transformed into their embeddable counterparts.
- **Checklist Heuristics:** Added boundary tests for character limits and item counts to ensure proper classification between checklists and plain text widgets.

### ✨ Result:
Increased the total number of tests in `utils/smartPaste.test.ts` from 18 to 31, ensuring that the critical smart-paste logic is robust against regressions.

---
*PR created automatically by Jules for task [1333681013903441932](https://jules.google.com/task/1333681013903441932) started by @OPS-PIvers*

### CODE REVIEW

File: utils/smartPaste.test.ts [hunk 1]
```diff
@@ -1,8 +1,13 @@
 import { describe, it, expect } from 'vitest';
 import { detectWidgetType } from './smartPaste';
-import { EmbedConfig, ChecklistConfig } from '../types';
+import { EmbedConfig, ChecklistConfig, StickerConfig } from '../types';
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply utils/smartPaste.test.ts]: Fixed in the latest commit — the relative `'../types'` import has been changed to the `'@/types'` path alias.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by OPS-PIvers — COMMENTED** [review]
## Automated Code Review

**Date:** 2026-04-21

### Summary
Expands `utils/smartPaste.test.ts` from 18 to 31 tests, covering input sanitization (empty/whitespace), share-link URL variants (boards, quizzes, assignments — both full URLs and bare domains), image/sticker detection across extensions and URL parameters, Google Sheets/Forms embed transforms, and checklist vs. text classification boundary cases. Low-risk, test-only change.

### Code Quality
- ✓ Tests follow existing patterns — `describe` grouping, `result?.action === 'x'` narrowing, explicit `throw new Error(...)` on unexpected branches.
- ✓ `describe('checklist edge cases', ...)` nested block correctly groups the boundary tests.
- ✓ Import path now uses the `@/types` alias (addressed in latest commit).

### Widget Standards
✓ Not applicable.

### Admin Config Standards
✓ Not applicable.

### Firestore & Security
✓ No rules changes or security-sensitive code.

### Regression Risk
✓ Low. Test-only addition with no production code changes. Confirmed all 31 tests pass locally after the alias fix.

### Missing Tests
- ✓ New coverage is comprehensive: sticker URL variants (8 cases), checklist heuristics (5 boundary cases), share URL bare-domain variants (2 cases), assignment share URLs (2 cases).
- ⚠ No test for `detectWidgetType` with leading/trailing whitespace on otherwise-valid URLs (e.g. `'  https://youtube.com/...  '`) — might be worth confirming trim behavior.

### Overall
**Merge readiness:** Ready
Clean test-only PR that meaningfully improves `smartPaste` regression protection. Ready to merge once CI is green.

---
_This review was generated automatically. All findings should be verified by a human before merging._

---
_Generated by [Claude Code](https://claude.ai/code)_
